### PR TITLE
WIP for add_to_view

### PR DESCRIPTION
**Pull request by jordansread**
Keeping this open

### CODE REVIEW

File: R/config.R [hunk 1]
```diff
@@ -107,12 +92,12 @@ formal_names <- function(type) {
   if (type == "par") {
     formals <- names(par(no.readonly=TRUE))
   } else {
-    func <- tryCatch(getFromNamespace(overrides[[type]], "graphics"), error=function(e){})
-    
-    if (!is.null(func)) {
-      formals <- names(formals(func))
+    funs <- function_defaults(type, out='def.funs')
```
Comment (jiwalker-usgs) [inline R/config.R:90]: I'm not quite sure what's going on there, but it is probably reasonable

File: R/add_to_view.R [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
+#' @param object a gsplot object
+#' @param call.args a named list of function calls
+#' @param side a numeric vector of side(s) appropriate for creating a view name
+#' @return a modified object with the function added to the proper view
+#' @keywords internal
+add_to_view <- function(object, call.args, side){
+  view.name <- as.view_name(side)
+  new.view <- !view.name %in% view_names(object)
+  
+  if (new.view)
```
Comment (lindsayplatt) [inline R/add_to_view.R]: {} please

File: R/function_defaults.R [hunk 3]
```diff
@@ -1,15 +1,30 @@
-
+#' get the defaults for a function name 
+#' 
+#' parent package for function, and default functions for capturing arguments
+#' 
+#' @param fun.name a chacter vector (length 1) for the function name
+#' @param out the exports you want (defaults to 'package' & 'def.funs')
+#' @return a list if more than one output is selected, unlisted if length one
 function_defaults <- function(fun.name, out=c('package','def.funs')){
```
Comment (lindsayplatt) [inline R/function_defaults.R:9]: add overrides here

File: R/function_defaults.R [hunk 4]
```diff
@@ -1,15 +1,30 @@
...
+  stopifnot(length(fun.name) == 1)
   funs <- list('points'=
```
Comment (lindsayplatt) [inline R/function_defaults.R]: pull this out

**Comment by jordansread** [comment]
This should be good to go now

**Comment by jordansread** [comment]
merging, since we peer reviewed this.

**Pull request merged**